share/dkg/pedersen/dkg.go

```diff
@@ -164,7 +178,18 @@ func NewDistKeyHandler(c *Config) (*DistKeyGenerator, error) {
 		canIssue = true
 	} else if !isResharing && newPresent {
 		// fresh DKG case
-		secretCoeff := c.Suite.Scalar().Pick(c.Suite.RandomStream())
+		var randomStream cipher.Stream
+		// empty reader, use entropy from crypto/rand
+		if c.Reader == nil {
+			randomStream = random.New()
+		} else { // use the reader they gave, alone or combined
+			if c.UserReaderOnly {
+				randomStream = random.New(c.Reader)
+			} else {
+				randomStream = random.New(c.Reader, rand.Reader)
+			}
+		}
+		secretCoeff := c.Suite.Scalar().Pick(randomStream)
```

Comment:
Just a minor nitpick regarding style: Can't we initialize `randomStream` by default with `random.New()` and then re-initalize it depending on the options the user provided? Meaning something like that:

```
randomStream := random.New()
if c.Reader != nil && c.UserReaderOnly {
    randomStream = random.New(c.Reader)
}
if c.Reader != nil && !c.UserReaderOnly {
    randomStream = random.New(c.Reader, rand.Reader)
}
```